src/Infrastructure/LeanCode.Serialization/JsonConverters/JsonLaxDateTimeOffsetConverter.cs

```diff
@@ -0,0 +1,67 @@
+using System.Buffers;
+using System.Globalization;
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace LeanCode.Serialization;
+
+public class JsonLaxDateTimeOffsetConverter : JsonConverter<DateTimeOffset>
+{
+    private static readonly UTF8Encoding encoder = new(false);
+    private const int maxBufferSize = 128;
+    private const int maxDateTimeOffsetBufferSize = 32;
+    private readonly string format = "yyyy'-'MM'-'dd'T'HH':'mm':'ss.fffzzz";
+
+    public override DateTimeOffset Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        Span<char> source = stackalloc char[0];
+
+        if(reader.HasValueSequence)
+        {
+            var valueLength = (int)reader.ValueSequence.Length;
+            if (valueLength > maxBufferSize)
+            {
+                throw new JsonException($"Input too long ({valueLength})");
+            }
+            Span<char> buffer = stackalloc char[valueLength];
+            int charsRead = reader.CopyString(buffer);
+            source = buffer.Slice(0, charsRead);
+        }
+        else
+        {
+            var value = reader.ValueSpan;
+            if (value.Length > maxBufferSize)
+            {
+                throw new JsonException($"Input too long ({value.Length})");
+            }
+            Span<char> buffer = stackalloc char[value.Length];
+            int charsRead = reader.CopyString(buffer);
+            source = buffer.Slice(0, charsRead);
+        }
+
+        if (DateTimeOffset.TryParse(source, CultureInfo.InvariantCulture, out DateTimeOffset date))
+        {
+            var roundedTicks = (date.Ticks + TimeSpan.TicksPerMillisecond / 2) / TimeSpan.TicksPerMillisecond * TimeSpan.TicksPerMillisecond;
+            return new DateTimeOffset(roundedTicks, date.Offset);
+        }
+
+        throw new JsonException("Invalid format");
+    }
+
+    public override void Write(Utf8JsonWriter writer, DateTimeOffset value, JsonSerializerOptions options) {
+        Span<char> buffer = stackalloc char[maxDateTimeOffsetBufferSize];
+        var success = value.TryFormat(buffer, out var written, format, CultureInfo.InvariantCulture);
+
+        if(value.Offset == TimeSpan.Zero)
```

Comment:
I think it's okay to leave `+00:00` for UTC DTOs? I'll need to check that one though.
Also you don't need to null-terminate as long as you slice properly.